### PR TITLE
WPF - Add simplified resize hack (disabled by default)

### PR DESCRIPTION
**Pull request by Wupval**
**Fixes:**
   - Fixes issue 4274

**Summary:**
   - Re-Added the resize hack for issue 2779, as the issue still occurs. Disabled by default as mentioned by @amaitland in https://github.com/cefsharp/CefSharp/discussions/4274#discussioncomment-3946056

**Changes:**
   - Re-Added the resize hack for issue 2779
   - Ported changes to match master (Execution of the resize hack has been moved to OnBrowserWasHidden)
      
**How Has This Been Tested?**  
   - This was tested with the CefSharp minimal example, creating multiple browsers (at least 6 are needed) and toggling the visibility of the browsers.
   - OS: Win10, x64

**Types of changes**
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Updated documentation

### CODE REVIEW

File: CefSharp.Wpf/ChromiumWebBrowser.cs [hunk 1]
```diff
@@ -1759,11 +1795,26 @@ private void OnWindowStateChanged(object sender, EventArgs e)
...
-        protected virtual void OnBrowserWasHidden(bool hidden)
+        protected virtual async void OnBrowserWasHidden(bool hidden)
         {
             if (browser != null)
             {
                 browser.GetHost().WasHidden(hidden);
+
+                if (hidden)
+                {
+                    if (EnableResizeHackForIssue2779)
```
Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs:1806]: This if statement should be moved up so the whole set of conditions is only evaluated if hack enabled.

File: CefSharp.Wpf/ChromiumWebBrowser.cs [hunk 2]
```diff
@@ -2696,5 +2747,48 @@ public IBrowser GetBrowser()
...
+                    {
+                        action();
+                    });
+                }
+            }
+        }
+
+        private async Task ResizeHackForIssue2779()
+        {
+            if (EnableResizeHackForIssue2779)
```
Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs:2771]: Remove this if statement, method needs to only be called if enabled.

File: CefSharp.Wpf/ChromiumWebBrowser.cs [hunk 3]
```diff
@@ -1759,11 +1795,26 @@ private void OnWindowStateChanged(object sender, EventArgs e)
...
+                if (hidden)
+                {
+                    if (EnableResizeHackForIssue2779)
+                    {
+                        resizeHackForIssue2779Enabled = true;
+                    }
+                }
+                else
+                {
+                    await CefUiThreadRunAsync(async () =>
```
Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs:1813]: Let's remove the async modifier from the method and discard the Task returned from the method. 

There's no continuation.

**Review by amaitland — CHANGES_REQUESTED** [review]
Comments inline.

**Comment by Wupval** [comment]
@amaitland Is there an error in the code or any other reason the build seems to always fail?

**Comment by amaitland** [comment]
Build failure is unrelated to this PR.

**Comment by amaitland** [comment]
Follow up commit https://github.com/cefsharp/CefSharp/commit/0131dd8a331a3f500cf1c06da2b85307e9fc6de5

Properties have been renamed, no longer reference issue #2779, directly reference the upstream issue.

**Comment by amaitland** [comment]
This is a greatly simplified version of the resize hack originally used. If anyone experiences any issues then I suggest you override `OnBrowserWasHidden` and `OnPaint` to implement your own version.

Original version for reference https://github.com/cefsharp/CefSharp/commit/9140c08a34e113b8865a80a7ac9f8b4aa19631c4

**Pull request merged**